src/types/index.ts

```diff
@@ -1,46 +1,408 @@
 
-interface StopGroup {}
+type GroundPlacesObject = Record<Gpuid, StopGroup | StopCluster>;
 
-interface StopCluster {}
+type Gpuid = string;
 
-interface StopGroupProperties {}
+type StopGroupGpuid = Gpuid;
 
-interface StopClusterProperties {}
+type StopClusterGpuid = Gpuid;
 
-interface Filters {}
+// -----------------------------------------------------------------------------
+//  ------------------------------- INTERFACES ---------------------------------
+// -----------------------------------------------------------------------------
 
-interface GroundPlacesDiff {}
+interface StopGroup {
+  country_code: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType.GROUP;
+  childs: SegmentProviderStop[];
+  serviced: Serviced;
+  has_been_modified?: boolean;
+  warning?: boolean;
+  is_latest?: boolean;
+}
 
-type Gpuid = string;
+interface StopCluster {
+  country_code: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType.CLUSTER;
+  childs: StopGroupGpuid[];
+  serviced: Serviced;
+  unique_name?: string;
+  has_been_modified?: boolean;
+  warning?: boolean;
+  is_latest?: boolean;
+}
 
-type StopGroupGpuid = Gpuid;
+interface StopGroupInfos {
+  countryCode: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType;
+  serviced: Serviced;
+}
 
-type StopClusterGpuid = Gpuid;
+interface StopClusterInfos {
+  countryCode: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType;
+  createdFromStopGroupGpuid: StopGroupGpuid;
+  serviced: Serviced;
+}
+
+interface StopGroupProperties {
+  name?: string;
+  longitude?: number;
+  latitude?: number;
+}
+
+interface StopClusterProperties {
+  name?: string;
+  longitude?: number;
+  latitude?: number;
+}
+
+interface GroundPlaceGenerated {
+  id: StopGroupGpuid | StopClusterGpuid;
+  name: string;
+  latitude: number;
+  longitude: number;
+  countryCode: CountryCode;
+  type: GroundPlaceType;
+  ancestorId?: string;
+}
+
+interface SegmentProviderStop {
+  unique_name: null;
+  company_name: string;
+  name: string;
+  latitude: number;
+  serviced: Serviced;
+  company_id: number;
+  longitude: number;
+  id: string;
+}
+
+interface GroundPlacesDiff {
+  [gpuid: string]: {
+    type: ActionType;
+    into?: string;
+  };
+}
+
+// TODO: Implement these interfaces
+
+interface AutoComplete {}
+
+interface Filters {}
+
+// -----------------------------------------------------------------------------
+//  --------------------------------- ENUMS ------------------------------------
+// -----------------------------------------------------------------------------
+
+enum ActionType {
+  CREATE = 'create',
+  MOVE = 'move',
+  UPDATE = 'update',
+  DELETE = 'delete',
+}
 
 enum GroundPlaceType {
-  Group = 'group',
-  Cluster = 'cluster',
+  GROUP = 'group',
+  CLUSTER = 'cluster',
+}
+
+enum Serviced {
```

Comment:
No need for an enum to replace a bool...

Reply:
Serviced is not a boolean typed, this return a string, so I can't type like that `serviced: boolean`